test/libsolidity/Metadata.cpp

```diff
@@ -400,6 +400,55 @@ BOOST_AUTO_TEST_CASE(metadata_revert_strings)
 	BOOST_CHECK_EQUAL(metadata["settings"]["debug"]["revertStrings"], "strip");
 }
 
+BOOST_AUTO_TEST_CASE(metadata_optimiser_sequence)
+{
+	char const* sourceCode = R"(
+		pragma solidity >=0.0;
+		contract test {
+		}
+	)";
+
+	vector<tuple<string, string>> sequences =
+	{
+		// { "<optimizer sequence>", "<optimizer cleanup sequence>" }
+		{ "", "" },
+		{ "", "fDn" },
+		{ "dhfoDgvulfnTUtnIf", "" },
+		{ "dhfoDgvulfnTUtnIf", "fDn" }
+	};
+
+	auto check = [sourceCode](string const& optimizerSequence, string const& optimizerCleanupSequence)
+	{
+		OptimiserSettings optimizerSettings = OptimiserSettings::minimal();
+		optimizerSettings.runYulOptimiser = true;
+		optimizerSettings.yulOptimiserSteps = optimizerSequence;
+		optimizerSettings.yulOptimiserCleanupSteps = optimizerCleanupSequence;
+		CompilerStack compilerStack;
+		compilerStack.setSources({{"", std::string(sourceCode)}});
```

Comment:
```suggestion
		compilerStack.setSources({{"", string(sourceCode)}});
```
Also, won't an implicit conversion to `string` kick in here?

Reply:
The test setup is a copy-paste from another text, and looking at the file now, there's a ton of `std::string` usage, so not sure if I should include such changes in this PR. Also yes, a `char*` is implicitly convertible to a `string` :)

Reply:
ok. Fine then. I think we should fix individual cases we find but yeah, redoing the whole file would add too much noise. Not that important anyway.